arches_lingo/views/api/concepts.py

```diff
@@ -0,0 +1,87 @@
+from http import HTTPStatus
+
+from django.core.paginator import Paginator
+from django.utils.decorators import method_decorator
+from django.utils.translation import gettext as _
+from django.views.generic import View
+
+from arches.app.models.system_settings import settings
+from arches.app.utils.decorators import group_required
+from arches.app.utils.response import JSONErrorResponse, JSONResponse
+
+from arches_lingo.models import VwLabelValue
+from arches_lingo.concepts import ConceptBuilder
+
+
+@method_decorator(
+    group_required("RDM Administrator", raise_exception=True), name="dispatch"
+)
+class ConceptTreeView(View):
+    def get(self, request):
+        builder = ConceptBuilder()
+        data = {
+            "schemes": [builder.serialize_scheme(scheme) for scheme in builder.schemes]
+        }
+        return JSONResponse(data)
+
+
+@method_decorator(
+    group_required("RDM Administrator", raise_exception=True), name="dispatch"
+)
+class ValueSearchView(ConceptTreeView):
+    def get(self, request):
+        term = request.GET.get("term")
+        max_edit_distance = request.GET.get(
+            "maxEditDistance", self.default_sensitivity()
+        )
+        exact = request.GET.get("exact", False)
+        page_number = request.GET.get("page", 1)
+        items_per_page = request.GET.get("items", 25)
+
+        if exact:
+            concept_query = VwLabelValue.objects.filter(value=term).order_by(
+                "concept_id"
+            )
+        elif term:
+            try:
+                concept_query = VwLabelValue.objects.fuzzy_search(
+                    term, max_edit_distance
+                )
+            except ValueError as ve:
+                return JSONErrorResponse(
+                    title=_("Unable to perform search."),
+                    message=ve.args[0],
+                    status=HTTPStatus.BAD_REQUEST,
+                )
+        else:
+            concept_query = VwLabelValue.objects.all().order_by("concept_id")
+        concept_query = concept_query.values_list("concept_id", flat=True).distinct()
+
+        paginator = Paginator(concept_query, items_per_page)
+        page = paginator.get_page(page_number)
+
+        data = []
+        if page:
```

Comment:
nit: instead of control flow based on presence/absence of a pagintor page, can it based on the data fed into the paginator?

Reply:
The ConceptBuilder does queries to build a tree (or gets the cached one). If there are no search results, there's no need to build a tree, we can just quickly return empty. That was my thinking. I'm not exactly sure what you're suggesting as an alternative. I can clarify this to deindent a little bit.

Reply:
ya, I've not put a breakpoint to see if this holds up but

`concept_query = concept_query.values_list("concept_id", flat=True).distinct()`

returns something that will either have a length or not? 

Since `page_number` looks like it will always have a value, the only time there shouldn't be a page is when there's not data. Admittedly this is my assumption.

So instead of `if page:`, can it be `if len(concept_query): ( or whatever its new name is )`?

Reply:
Yeah I can signpost better. But we can't evaluate the queryset too early or we'll unnecessarily fetch lots of objects before doing the pagination to slim down the results.

Reply:
In case the implicit boolenness of the Page object is too opaque, I'll change it to `if not len(page)` to clarify it's an iterable of objects.

Reply:
Actually, it saves one query in the no-data path to check `page.count`, which seems like something I might ask on the Django forum about because `len()` should work the same way. So I'll do that.

Reply:
Oops, this should be .count on the paginator, not the page.